nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,2575 @@
+      ],
+      "execution_count": null,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stderr",
+          "text": [
+            "Using custom data configuration default\n",
+            "Reusing dataset emotion (/root/.cache/huggingface/datasets/emotion/default/0.0.0/348f63ca8e27b3713b6c04d723efe6d824a56fb3d1449794716c0f0296072705)\n"
+          ]
+        },
+        {
+          "output_type": "display_data",
+          "data": {
+            "application/vnd.jupyter.widget-view+json": {
+              "model_id": "b39348e38bfa4ff6a7266b023f0c58a8",
+              "version_minor": 0,
+              "version_major": 2
+            },
+            "text/plain": [
+              "  0%|          | 0/3 [00:00<?, ?it/s]"
+            ]
+          },
+          "metadata": {}
+        },
+        {
+          "output_type": "stream",
+          "name": "stderr",
+          "text": [
+            "Loading cached processed dataset at /root/.cache/huggingface/datasets/emotion/default/0.0.0/348f63ca8e27b3713b6c04d723efe6d824a56fb3d1449794716c0f0296072705/cache-7161aca97360dca6.arrow\n",
+            "Loading cached processed dataset at /root/.cache/huggingface/datasets/emotion/default/0.0.0/348f63ca8e27b3713b6c04d723efe6d824a56fb3d1449794716c0f0296072705/cache-51b5751515a95f0f.arrow\n",
+            "Loading cached processed dataset at /root/.cache/huggingface/datasets/emotion/default/0.0.0/348f63ca8e27b3713b6c04d723efe6d824a56fb3d1449794716c0f0296072705/cache-3ad9ee5140e9d78b.arrow\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RQ24-mWS6elF"
+      },
+      "source": [
+        "The dataset is already split into 16,000 train, 2,000 validation and 2,000 test samples. To investigate the effectiveness of the GPT3Mix augmentation method, we will use only 10 samples per class as a train set."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "9YI6-pZdSqnI",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "outputId": "60d03419-bc4b-4f5c-ca74-3c4ebe633043"
+      },
+      "source": [
+        "# select 10 random train samples from each of the three emotions\n",
+        "# sadness (0), joy (1), love (2), anger (3), fear (4), surprise (5)\n",
+        "joy_train_samples = emotion_ds['train'].filter(lambda e: e['label'] == 1).select(range(10))\n",
+        "anger_train_samples = emotion_ds['train'].filter(lambda e: e['label'] == 3).select(range(10))\n",
+        "surprise_train_samples = emotion_ds['train'].filter(lambda e: e['label'] == 5).select(range(10))\n",
+        "\n",
+        "# map emotions to integers for labeling\n",
+        "# joy (0), anger (1), surprise (2)\n",
+        "def map_emotions(example):\n",
+        "  example['label'] = example['label']//2\n",
```

Comment:
I'm not quite sure what you're planning to do here, are you using this math operator to map (1,3,5) to (1,2,3) ? If so, this feels a bit hacky, might be better to do this explicitly